src/cache/cache.rs

```diff
@@ -125,7 +125,19 @@ impl CacheWrite {
     pub fn put_object<T>(&mut self, name: &str, from: &mut T, mode: Option<u32>) -> Result<()>
         where T: Read,
     {
-        let opts = FileOptions::default().compression_method(CompressionMethod::Deflated);
+        let method = if let Ok(ref method) = env::var("SCCACHE_COMPRESSION_METHOD") {
+            match method.to_lowercase().trim() {
+                "deflate" => CompressionMethod::Deflated,
+                "none" => CompressionMethod::Stored,
+                _ => {
+                    warn!("Unknown compression method {} using 'deflate'.", method);
```

Comment:
This would silently fail on typos etc. Maybe it's better to return `Err` to show the user that something has gone wrong